### ISSUE TITLE
This method creates a model with a 100% memory leak loop using model. fit()

### ISSUE THREAD

**Issue by phpYj** [keras-team/keras#20552]
I have tried various methods, but the memory is definitely leaking, it seems that the release of memory cannot keep up. Through the logs, it can be found that there is periodic memory recycling, but with the increase of time, there is still a clear upward trend

Please find the [gist](https://colab.sandbox.google.com/gist/Venkat6871/30fb12f2188a7826e2c649bbd945dbda/80753_tf_2-18-0-nightly-v.ipynb) here for your reference.

`import gc

import keras
import numpy as np
import psutil
from keras.optimizers import Adam
from keras.layers import Dense, Dropout, Input, LSTM
import tensorflow as tf
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import MinMaxScaler
from tensorflow.keras.models import Sequential, load_model
from tensorflow.keras.callbacks import EarlyStopping, ModelCheckpoint
import time
import json

num_samples = 6
num_features = 3
num_classes = 4
epochs = 50
batch_size = 2
identifier = "test_model"
num_iterations = 500  

def build_model(X, num_classes):
    model = Sequential()
    model.add(Input(shape=(X.shape[1], X.shape[2])))
    model.add(LSTM(16, return_sequences=True))
    model.add(LSTM(16))
    model.add(Dropout(0.4))
    model.add(Dense(8, activation='tanh'))
    model.add(Dense(num_classes, activation='softmax'))

    model.compile(loss='categorical_crossentropy', optimizer=Adam(), metrics=['categorical_accuracy'])
    return model

data_X = np.random.rand(num_samples, num_features)
data_Y = np.random.randint(0, num_classes, size=(num_samples, 1))  

data_Y = np.eye(num_classes)[data_Y.flatten()]  
print(type(data_X))

scaler = MinMaxScaler()
data_X_scaled = scaler.fit_transform(data_X)

train_X, test_X, train_Y, test_Y = train_test_split(data_X_scaled, data_Y, train_size=0.6, random_state=42)

train_X = np.expand_dims(train_X, axis=1)
test_X = np.expand_dims(test_X, axis=1)

best_loss = np.inf
best_model_data = None
for iteration in range(num_iterations):
   
    tf.keras.backend.clear_session()
    tf.compat.v1.reset_default_graph()
    model = build_model(train_X, num_classes)
 
    model_name = f"model_{iteration}"
    early_stopping = EarlyStopping(monitor='loss', patience=10, verbose=0, restore_best_weights=True)
    print(f"Iteration {iteration + 1}/{num_iterations}")
    process = psutil.Process()
    mem_info = process.memory_info()
    print(f"start Current memory usage: {mem_info.rss / (1024 * 1024):.2f} MB")  # RSS - Resident Set Size
    try:
        history = model.fit(train_X, train_Y, epochs=epochs, batch_size=batch_size, shuffle=True,
                            validation_data=(test_X, test_Y), verbose=0)
        current_loss = history.history['loss'][-1]
        print(f"Training model: {model.name}")
    
        del model
        tf.keras.backend.clear_session()
        gc.collect()
    except Exception as e:
        print("err:", e)
    finally:
        process = psutil.Process()
        mem_info = process.memory_info()
        print(f"end Current memory usage: {mem_info.rss / (1024 * 1024):.2f} MB")  # RSS - Resident Set Size

print("end！")

if best_model_data:
    model_json = best_model_data["model_architecture"]
    model_weights = json.loads(best_model_data["model_weights"], object_hook=lambda d: np.array(d))
    model = tf.keras.models.model_from_json(model_json)
    model.set_weights(model_weights)
    model.compile(loss='categorical_crossentropy', optimizer=Adam(), metrics=['categorical_accuracy'])
    print("ok")
else:
    print("not found")`

**Comment by fchollet** [comment]
Is this reproducible with a different backend?

TF has some well known memory leak issues, but there isn't anything we can do about it on the Keras side, unless the issue is also visible in other backends (in which case the issue is with the Keras codebase).

**Comment by RHTNT** [comment]
It is a tensorflow bug. I searched through virtually all issues related to this bug and basically there are one or maybe two approaches apprently:
1) Run your code in a separate process, retrieve the output somehow and kill the process. That way the memory will be freed.
2) Maybe @https://github.com/tensorflow/tensorflow/issues/80753#issuecomment-2559339312 works for you.
